timemachine/md/hrex.py

```diff
@@ -152,6 +152,31 @@ def symmetrize(x):
     return 1 / (1 - mu_2)
 
 
+def get_samples_by_iter_by_replica(
+    samples_by_state_by_iter: List[List[Samples]], replica_idx_by_state_by_iter: List[List[ReplicaIdx]]
+) -> List[List[Samples]]:
+    """Permute and reshape samples returned by :py:func:`run_hrex` having shape
+
+        (hrex_iter, state_idx) -> samples
+
+    to a more convenient form for analyzing the trajectories of individual replicas
+
+        (replica_idx, hrex_iter) -> samples
+    """
+
+    def inverse_permutation(p):
+        return [i for i, _ in sorted(enumerate(p), key=lambda t: t[1])]
```

Comment:
(alternatively, `np.argsort(p)`)

Reply:
Changed in https://github.com/proteneer/timemachine/pull/1217/commits/add35fb05fab9e903867420d3d956d2041328530. Not sure why I had been avoiding numpy here.